File: ruby_small_problems/easy_4/5.rb
# searches for all multiples of 3 or 5 that lie between 1 and some other number
# computs the sum of those multiples

def multisum(num)
  (1..num).select{ |x| (x % 3 == 0) || (x % 5 == 0) }.sum
end

puts multisum(3) == 3
puts multisum(5) == 8
puts multisum(10) == 33
puts multisum(1000) == 234168

File: 210/lecture/practice_problems/logic_and_flow_control/5.js
function isPrime(number) {
  if (number <= 1 || (number > 2 && number % 2 == 0)) {
    return false; 
  }

  for (i = 2; i < number; i += 1) {
    if (number % i == 0) {
      return false; 
    }
  }

  return true;
}

console.log(isPrime(1));   // false
console.log(isPrime(2));   // true
console.log(isPrime(3));   // true
console.log(isPrime(0));   // false
console.log(isPrime(43));  // true
console.log(isPrime(55));  // false
